lib/verifier/hathifiles_database_verifier.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require "zlib"
+
+require_relative "../verifier"
+require_relative "../derivatives"
+
+module PostZephirProcessing
+  class HathifilesDatabaseVerifier < Verifier
+    attr_reader :current_date
+
+    # Does an entry exist in hf_log for the hathifile?
+    # Can pass a path or just the filename.
+    def self.has_log?(hathifile:)
+      PostZephirProcessing::Services[:database][:hf_log]
+        .where(hathifile: File.basename(hathifile))
+        .count
+        .positive?
+    end
+
+    # Count the number of entries in hathifiles.hf
+    def self.db_count
+      PostZephirProcessing::Services[:database][:hf].count
+    end
+
+    def run_for_date(date:)
+      @current_date = date
+      verify_hathifiles_database_log
+      verify_hathifiles_database_count
+    end
+
+    private
+
+    def verify_hathifiles_database_log
+      # File missing? Not our problem, should be caught by earlier verifier.
+      if File.exist?(update_file)
```

Comment:
pattern from hathifiles listing should help here (plus using a Derivative class)